tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6621,13 +6613,45 @@ Status ConvertResize(OpConverterParams* params) {
   // Calculate output dimensions.
   // Given input dimensions [N, C, H, W] and output size [H_out, W_out],
   // output dimensions equals [N, C, H_out, W_out]
-  nvinfer1::Dims output_dimensions;
-  output_dimensions.nbDims = inputs_tensor->getDimensions().nbDims;
-  for (int i = 0; i < output_dimensions.nbDims; ++i) {
-    output_dimensions.d[i] = inputs_tensor->getDimensions().d[i];
+  const auto inputs_dims = inputs_tensor->getDimensions();
+  nvinfer1::Dims output_shape_dims;
+  ITensorProxyPtr output_shape_tensor;
+  if (params->use_implicit_batch || HasStaticShape(inputs_dims)) {
```

Comment:
Shall we remove params->use_implicit_batch here?

Reply:
This branch should also run when running in implicit batch mode, as in that case the output dimensions need to be explicitly set as well. However if you think it's cleaner -- I could change the conditionals to be more "descriptive" by checking for whether `output_shape_dims` or `output_shape_tensor` have been set in the if statement above, and run the corresponding branch in either case.

Reply:
I have two questions:
(1) the block of code inside this if-branch also works and executes when (use_implicit_batch==false && HasStaticShape(inputs_dims)) before your PR, see [code](https://github.com/tensorflow/tensorflow/blob/2825dbfe950cd514828da4bfed81a2402d779b35/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc#L6609-L6612). Am I right here? If not, what are the reason that this code don't work for use_implicit_batch=false? I couldn't see that...
(2) This is not introduced by your PR, but when I read [this code](https://github.com/tensorflow/tensorflow/blob/2825dbfe950cd514828da4bfed81a2402d779b35/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc#L6618-L6619), I think it has a problem for use_implicit_batch=true as we remove the batch dimension and should only have three dimensions in the input tensor.  Do you agree?

Reply:
Sure, so:
(1) When running in implicit batch mode, all non-batch dimensions are known at build time. So that is enough to be able to calculate the output shape dimensions. So the implicit batch check there acts as a flag that allows this dimension calculation to happen at build time. In any case, this logic is slightly modified in the commit I'm pushing soon for dynamic output size support.
(2) Kind of -- So in implicit batch mode, during build time, the batch dimension is skipped and all build-time calculations must happen ignoring that dimension. However at run-time, the batch dimension does exist obviously, so any runtime computations must take the N batch dimension into account. So the transpose layer that you pointed out must be configured with how the layer will operate at runtime, and so the transpose dims must be configured taking the batch dimension in account.

Reply:
We discussed this offline, I think the current version now addresses your concern with how implicit batch is checked for.